test/functional/sqlsrv/MsHelper.inc

```diff
@@ -94,14 +93,23 @@ class BindParamOption
         $direction = null;
         $phpType = null;
         $sqlType = null;
-        if ($this->direction) {
-            $direction = constant($this->direction);
+        if ($this->direction) {          
+            if (in_array($this->direction, array(SQLSRV_PARAM_IN, SQLSRV_PARAM_OUT, SQLSRV_PARAM_INOUT))) {
+                $direction = constant($this->direction);
+            } else {
+                echo "BindParamOption: invalid direction for parameter!\n";
+            }
         }
         if ($this->phpType) {
-            $phpType = constant($this->phpType);
+            try {
+                $phpType = constant($this->phpType);
+            } catch (Exception $e) {
+                // there's something wrong with the input php type 
+                echo $e->getMessage();
+            }
```

Comment:
have you tried it? does it throw an exception when the constant is not found?

Reply:
yes, I've tried SQLSRV_PHPTYPE_FLOAT1 and SQLSRV_PHPTYPE_FLOAT(1), the first resulted in a PHP warning and the other a PHP Fatal error.